Source/WPEFramework/PluginServer.cpp

```diff
@@ -337,14 +337,18 @@ ENUM_CONVERSION_BEGIN(Core::ProcessInfo::scheduler)
                 State(ACTIVATION);
                 _administrator.StateChange(this);
 
-                Unlock();
```

Comment:
The Initialize should *not* be within the locked scope. This might cause severe deadlocks during startup.

Reply:
And continuing the revieuw, I think you did this to "lock" the launch of the LD_LIBRARY_PATH, which by using this lock is now done from the Initialize, but this impact "all" plugins. It would be sufficient to only protect the launch with a COM global lock. Than only "remote" instantiations will be locked and thus the scope is very minimal.